src/compiler/program.ts

```diff
@@ -4320,97 +4322,115 @@ export function createProgram(rootNamesOrOptions: readonly string[] | CreateProg
         }
     }
 
-    function getVersionForDeprecationDiagnostics(reportInvalidIgnoreDeprecations: boolean) {
-        const version = typeScriptVersion || versionMajorMinor;
+    function getTypeScriptVersion(): [version: Version, versionString: string] {

```

Comment:
This function is overcomplicated due to some old stuff I was doing. Will change.